src/Symfony/Component/ObjectMapper/Tests/ObjectMapperTest.php

```diff
@@ -651,4 +655,23 @@ public function testBugReportLazyLoadingPromotedReadonlyProperty()
         $this->assertSame('foo', $out->var1);
         $this->assertSame('bar', $out->b->var2);
     }
+
+    public function testClassMap()
+    {
+        $classMap = [
+            Cost::class => CostRequestView::class,
+            Quote::class => QuoteRequestView::class,
+        ];
```

Comment:
I see why it's working, I think we need a test in the framework bundle also as currently it won't work there (though the component's logic is okay).